activation/activation.go

```diff
@@ -368,6 +371,23 @@ func (b *Builder) waitForFirstATX(ctx context.Context) bool {
 	return true
 }
 
+func (b *Builder) updatePoetClients() {
+	poetClients := func() []PoetProvingServiceClient {
+		for {
+			select {
+			case clients := <-b.pendingPoetClients:
+				return clients
+			default:
+				return nil
+			}
+		}
+	}()
```

Comment:
It should pick up the last element available in the channel (the last update of poet clients). Currently, it picks up the first one (the oldest update).

Reply:
with atomic (or mutex locked variable also works) it will work exactly this way